src/WebJobs.Script.WebHost/Models/HostAssignmentContext.cs

```diff
@@ -44,6 +46,21 @@ public string ZipUrl
             }
         }
 
+        public bool TryGetMsiEndpoint(out string msiEndpoint)
+        {
+            msiEndpoint = null;
+            if (Environment.TryGetValue(EnvironmentSettingNames.MsiEndpoint, out msiEndpoint))
+            {
+                string msiSecret;
+                if (Environment.TryGetValue(EnvironmentSettingNames.MsiSecret, out msiSecret))
```

Comment:
Who sets this MsiSecret env var?

Reply:
This will be part of the rest of specialization payload. (from appservice)